python-lib/pagination.py

```diff
@@ -66,7 +67,11 @@ def update_next_page(self, data, response_links=None):
                 self.is_last_batch_empty = True
             return
         elif self.counting_key:
-            batch_size = len(data.get(self.counting_key, []))
+            extracted_data = get_value_from_path(data, self.counting_key.split("."), can_raise=False)
+            if extracted_data:
+                batch_size = len(get_value_from_path(data, self.counting_key.split("."), can_raise=False))
+            else:
+                batch_size = 0
             if batch_size == 0:
                 self.is_last_batch_empty = True
```

Comment:
this if is not needed `if batch_size == 0`: if extracted_data is True,` len(extracted_data)`>0, else, `batch_size = 0 `and `self.is_last_batch_empty` can be set to True
```suggestion
                batch_size = 0
                self.is_last_batch_empty = True
```

Reply:
Fixed in a973c70